Subsystems

EphraimSonnenberg opened the pull request:
- Added Intake Subsystem
- Added Arm Subsystems and IO. Not fully done.
- Updated to Phoenix 6 util and added skeleton for new autologging stuff, Made ArmIOSim and ArmIOTalonFX more detailed, and Did debugging and logging for arm subsystem.
- Added Elevator Subsystem with Boden, Merge branch 'master' into subsystems, Added Wrist Subsystem, Fingeys Subsystem, and added Test Code for Wrist, and Refactored arm code, moved to autolog, separated arm joints, removed talon from sim, made sim more accurate.
- Finished arm code
- Merge branch 'master' into subsystems, Removed formator because it was preventing us from building, and NON FUNCTIONAL - Broken Elevator Code + Most of Wrist Subsystem
- Added Elevator Subsystem Code, Finished elevator and added prototype arm code, and NON FUNCTIONAL - Fingeys Code
- Finished Fingeys Subsystem and started Intake Subsysytem.
- Implemented arm constants class and bugfixed arm, Finished up fingeys (for realsies this time), Finished up robot state logging and added test commands, Finished up intake and added test bindings, Made real io init properly, Merge remote-tracking branch 'origin/master' into subsystems, Updated ChoreoLib, Add more accurate sim stats, Add more accurate sim stats, removed MotionMagic, and Fixed Intake Subsystem.
- Cleaned up subsystems, fixed accidental deletion when cleaning subsystems, and Added basic triggers to subsystems.
- Toesies Subsystem Added, NOT FUNCTIONING - Added Toesies and IntakeExtender Code, Intake extender done and toesies done also, and Added Current Limits.
- Setup 2d mechanism sim, Added ConfigureTestButtonBindings with LoggedTunableNumbers, and Merge branch 'subsystems' of https://github.com/FRC930/Robot2025 into subsystems.
- changed one line of code in fingeys.java, Fixed Fingeys Code, Added ToesiesConstants, and added intakeExtender 2D mechanism.

EphraimSonnenberg committed:
Added Intake Subsystem

EphraimSonnenberg committed:
Added Arm Subsystems and IO. Not fully done.

Gregadeaux committed:
Updated robotstate to render the krayon cad version of the robot in simulation

bodenaarhus committed:
finished arm io's and simulation code as well as added prototype class

Co-authored-by: Keegan Rubingh <KeeganRubingh@users.noreply.github.com>
Co-authored-by: NickTheFox99 <NickTheFox99@users.noreply.github.com>

KeeganRubingh committed:
Updated Phoenix 6 util and added skeleton for new autologging stuff

EphraimSonnenberg committed:
Made ArmIOSim and ArmIOTalonFX more detailed

KeeganRubingh committed:
Did debugging and logging for arm subsystem

Co-authored-by: EphraimSonnenberg <EphraimSonnenberg@users.noreply.github.com>

EphraimSonnenberg committed:
Added Elevator Subsystem with Boden

EphraimSonnenberg committed:
Added Wrist Subsystem, Intake Subsystem, Fingeys Subsystem, and added Test Code for Wrist.

KeeganRubingh committed:
Refactored arm code, moved to autolog, seperated arm joints, removed talon from sim, made sim more accurate

KeeganRubingh committed:
Finished arm code

KeeganRubingh committed:
Removed formatter because it was preventing us from building

EphraimSonnenberg committed:
NON FUNCTIONAL - Broken Elevator Code + Most of Wrist Subsystem

EphraimSonnenberg committed:
Added Elevator Subsystem Code

KeeganRubingh committed:
Finished elevator and added prototype arm code

EphraimSonnenberg committed:
NON FUNCTIONAL - Fingeys Code

EphraimSonnenberg committed:
Finished Fingeys Subsystem and started Intake Subsysytem.

KeeganRubingh committed:
Implemented arm constants class and bugfixed arm

KeeganRubingh committed:
Finished up fingeys (for realsies this time)

KeeganRubingh committed:
Finished up robot state logging and added test commands

KeeganRubingh committed:
Finished up intake and added test bindings

KeeganRubingh committed:
Made real io init properly

KeeganRubingh committed:
Updated ChoreoLib

KeeganRubingh committed:
Add more accurate sim stats

KeeganRubingh committed:
removed MotionMagic

bodenaarhus committed:
Fixed Intake Subsystem

KeeganRubingh committed:
Cleaned up subsystems

KeeganRubingh committed:
fixed accidental deletion when cleaning subsystems

KeeganRubingh committed:
Added basic triggers to subsystems

EphraimSonnenberg committed:
Toesies Subsystem Added

EphraimSonnenberg committed:
NOT FUNCTIONING - Added Toesies and IntakeExtender Code

bodenaarhus committed:
Intake extender done and toesies done also

EphraimSonnenberg committed:
Added Current Limits

KeeganRubingh committed:
Setup 2d mechanism sim

EphraimSonnenberg committed:
Added ConfigureTestButtonBindings with LoggedTunableNumbers

bodenaarhus committed:
changed one line of code in fingeys.java

bodenaarhus committed:
Fixed Fingeys Code

bodenaarhus committed:
Added ToesiesConstants

bodenaarhus committed:
added intakeExtender 2D mechanism

HarryOTO commented on src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -91,6 +141,25 @@ public RobotContainer() {
+        shoulder = new ArmJoint( new ArmJointIOTalonFX(new ShoulderConstants()));
+        elbow = new ArmJoint( new ArmJointIOTalonFX(new ElbowConstants()));
+
+        fingeys = null;
```
Does leaving fingeys null crash the code when a button is pressed? If fingeys subsystem is done it should probably be initiated.

HarryOTO commented on src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -91,6 +141,25 @@ public RobotContainer() {
+        fingeys = null;
+
+        //You'll have to initialize this yourself, since there's no CAN range marked.
+        intake = null;
```
This aswell.

HarryOTO commented on src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -128,6 +207,14 @@ public RobotContainer() {
+        fingeys = null;
+        intake = null;
+        toesies = null;
+        intakeExtender = null;
```
Are all of these meant to be left null? Add a comment for explanation of why these are null if so.

HarryOTO commented on src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -212,6 +301,17 @@ private void configureButtonBindings() {
+        .until(elbow.getNewAtSetpointTrigger().and(shoulder.getNewAtSetpointTrigger()))
+      )
+    
+    ).onFalse(elbow.getNewSetAngleCommand(0).alongWith(shoulder.getNewSetAngleCommand(0)));
```
Please test these angles in simulation and show me to verify.

HarryOTO commented on src/main/java/frc/robot/RobotState.java:
```diff
@@ -23,22 +29,66 @@ public class RobotState extends VirtualSubsystem {
+    SmartDashboard.putData("Mech2d",primaryMechanism2d);
+
+    // shoulder 18in
+    // elbow 15in
```
Maybe we should move this to a separate class but thats for later

HarryOTO commented on src/main/java/frc/robot/RobotState.java:
```diff
@@ -69,6 +119,10 @@ public Angle getWristTwist() {
 
+  public Angle getIntakeExtenderAngle() {
+    return intakeExtenderAngle;
+  }
```
?????

HarryOTO commented on src/main/java/frc/robot/RobotState.java:
```diff
@@ -92,7 +170,7 @@ private void visualize() {
                 new Transform3d(
                     new Translation3d(
-                        Meters.zero(), Meters.zero(), Inches.of(elevatorHeightTune.get())),
+                        Meters.zero(), Meters.zero(), this.elevatorHeight),
```
Please comment units or use util

HarryOTO commented on src/main/java/frc/robot/RobotState.java:
```diff
@@ -102,7 +180,7 @@ private void visualize() {
                     new Translation3d(),
                     new Rotation3d(
-                        Degrees.of(shoulderAngleTune.get()), Degrees.zero(), Degrees.zero())))
+                        this.shoulderAngle, Degrees.zero(), Degrees.zero())))
```
comment units or use util

HarryOTO commented on src/main/java/frc/robot/RobotState.java:
```diff
@@ -112,7 +190,7 @@ private void visualize() {
                     new Translation3d(),
                     new Rotation3d(
-                        Degrees.of(elbowAngleTune.get()), Degrees.zero(), Degrees.zero())))
+                        this.elbowAngle, Degrees.zero(), Degrees.zero())))
```
comment units or use util

HarryOTO commented on src/main/java/frc/robot/RobotState.java:
```diff
@@ -124,9 +202,15 @@ private void visualize() {
                     new Translation3d(),
                     new Rotation3d(
-                        Degrees.of(0), Degrees.of(wristTwistTune.get()), Degrees.zero())))
+                        Degrees.of(0), wristTwist, Degrees.zero())))
```
Comment units or use util

HarryOTO commented on src/main/java/frc/robot/subsystems/arm/ArmJoint.java:
```diff
@@ -0,0 +1,68 @@
+    m_armJointIO.setTarget(angle);
+  }
+
+  public Command getNewSetAngleCommand(double i) {
```
Please add units to javadoc

HarryOTO commented on src/main/java/frc/robot/subsystems/arm/ArmJoint.java:
```diff
@@ -0,0 +1,68 @@
+    });
+  }
+
+  public Trigger getNewAtSetpointTrigger() {
```
javadoc

HarryOTO commented on src/main/java/frc/robot/subsystems/elevator/Elevator.java:
```diff
@@ -0,0 +1,69 @@
+        this);
+  }
+
+  public Command getNewSetDistanceCommand(double i) {
```
Please add units to javadoc

HarryOTO commented on src/main/java/frc/robot/subsystems/elevator/Elevator.java:
```diff
@@ -0,0 +1,69 @@
+  public Command getNewSetDistanceCommand(double i) {
+    return new InstantCommand(
+        () -> {
+          setDistance(Meter.of((Meters.convertFrom(i, Inches))));
```
unnecessary conversions? Need a distance, do Inches.of(i) and convert from Distance class in TalonFXIO?

HarryOTO commented on src/main/java/frc/robot/subsystems/elevator/ElevatorIOSim.java:
```diff
@@ -0,0 +1,70 @@
+  }
+
+  @Override
+  public void setTarget(Distance target) {
```
explain that calcs volt from dist in javadoc pwease

HarryOTO commented on src/main/java/frc/robot/subsystems/elevator/ElevatorIOTalonFX.java:
```diff
@@ -0,0 +1,75 @@
+  public ElevatorIOTalonFX() {}
+
+
+  public static final double SPOOL_RADIUS = 1.0;//TODO: Set this
```
Right soooo. Is this safe?

HarryOTO replied:
also unit

EphraimSonnenberg replied:
We need to talk about this with Mechanical.

HarryOTO commented on src/main/java/frc/robot/subsystems/fingeys/Fingeys.java:
```diff
@@ -0,0 +1,54 @@
+        },
+        this);
+  }
+  public Command getNewSetVoltsCommand(double i) {
```
Please add units to javadoc

HarryOTO commented on src/main/java/frc/robot/subsystems/intake/Intake.java:
```diff
@@ -0,0 +1,51 @@
+        },
+        this);
+  }
+  public Command getNewSetVoltsCommand(double i) {
```
Please add units to javadoc

HarryOTO commented on src/main/java/frc/robot/subsystems/intake/IntakeConstants.java:
glad this one exists :> !!!

HarryOTO commented on src/main/java/frc/robot/subsystems/intake/IntakeIOSim.java:
```diff
@@ -0,0 +1,79 @@
+
+public class IntakeIOSim implements IntakeIO {
+  /** J3_KP makes the joint motor know what speed to run at */
+  public final double J3_KP = 1.0;
```
wth is J3

HarryOTO commented on src/main/java/frc/robot/subsystems/intake/IntakeIOTalonFX.java:
```diff
@@ -0,0 +1,72 @@
+  public IntakeIOTalonFX(int id,int CANrangeID) {
+    Motor = new TalonFX(id);
+    Request = new VoltageOut(0.0);
+    rangeSensor = new CANrange(CANrangeID);
```
Make constants in subsys maybe? For later tho not rn

HarryOTO commented on src/main/java/frc/robot/subsystems/intake/IntakeIOTalonFX.java:
```diff
@@ -0,0 +1,72 @@
+  private void configureTalons() {
+    TalonFXConfiguration cfg = new TalonFXConfiguration();
+    cfg.MotorOutput.NeutralMode = NeutralModeValue.Brake;
+    cfg.CurrentLimits.StatorCurrentLimit = 0;
```
these current limits are still set to 0 and enabled muppet

HarryOTO commented on src/main/java/frc/robot/subsystems/intake/IntakeIOTalonFX.java:
```diff
@@ -0,0 +1,72 @@
+  }
+
+  @Override
+  public Supplier<Distance> intakeSensorBro() {
```
mhm yup bro

HarryOTO commented on src/main/java/frc/robot/subsystems/intakeextender/IntakeExtender.java:
```diff
@@ -0,0 +1,69 @@
+
+  IntakeExtenderInputsAutoLogged loggedintakeExtender = new IntakeExtenderInputsAutoLogged();
+
+  //WRIST WEIGHT
```
????????????????????

HarryOTO commented on src/main/java/frc/robot/subsystems/intakeextender/IntakeExtender.java:
```diff
@@ -0,0 +1,69 @@
+    m_intakeextenderIO.setTarget(angle);
+  }
+  
+  public Command getNewIntakeExtenderTurnCommand(LoggedTunableNumber angle) {
```
Please add units to javadoc

HarryOTO commented on src/main/java/frc/robot/subsystems/intakeextender/IntakeExtender.java:
```diff
@@ -0,0 +1,69 @@
+      this);
+  }
+
+  public Command getNewIntakeExtenderTurnCommand(double i) {
```
Please add units to javadoc

HarryOTO commented on src/main/java/frc/robot/subsystems/intakeextender/IntakeExtenderConstants.java:
Dude really

EphraimSonnenberg replied:
really dude really 
dude dude really
really dude dude
dude really dude

HarryOTO commented on src/main/java/frc/robot/subsystems/intakeextender/IntakeExtenderIOSim.java:
```diff
@@ -0,0 +1,70 @@
+  private Voltage appliedVoltage = Volts.mutable(0.0);
+
+  private final ProfiledPIDController controller =
+      new ProfiledPIDController(2.4, 0.0, 0.0, new Constraints(100000, 361));
```
seems realistic

HarryOTO commented on src/main/java/frc/robot/subsystems/intakeextender/IntakeExtenderIOTalonFX.java:
```diff
@@ -0,0 +1,55 @@
+    TalonFXConfiguration cfg = new TalonFXConfiguration();
+    cfg.MotorOutput.NeutralMode = NeutralModeValue.Brake;
+    cfg.Voltage.PeakForwardVoltage = 7;
+    cfg.CurrentLimits.StatorCurrentLimit = 0;
```
Set to 0 and enabled btwww

HarryOTO commented on src/main/java/frc/robot/subsystems/wrist/WristIOTalonFX.java:
```diff
@@ -0,0 +1,55 @@
+    cfg.MotorOutput.NeutralMode = NeutralModeValue.Brake;
+    cfg.Voltage.PeakForwardVoltage = 7;
+    cfg.Voltage.PeakReverseVoltage = 7;
+    cfg.CurrentLimits.StatorCurrentLimit = 0;
```
These current limits are set to 0 and enabled. This will make the motor not run no matter what.

HarryOTO commented on src/main/java/frc/robot/subsystems/wrist/Wrist.java:
```diff
@@ -0,0 +1,69 @@
+        this);
+  }
+
+  public Command getNewWristTurnCommand(double i) {
```
Please add units to javadoc

HarryOTO commented on src/main/java/frc/robot/subsystems/toesies/ToesiesConstants.java:
Should not be used

HarryOTO commented on src/main/java/frc/robot/subsystems/toesies/Toesies.java:
```diff
@@ -0,0 +1,53 @@
+    m_ToesiesIO.setTarget(target);
+  }
+
+  public Command getNewSetVoltsCommand(double i) {
```
Please add units to javadoc

HarryOTO commented on src/main/java/frc/robot/subsystems/toesies/ToesiesIOTalonFX.java:
```diff
@@ -0,0 +1,59 @@
+  private void configureTalons() {
+    TalonFXConfiguration cfg = new TalonFXConfiguration();
+    cfg.MotorOutput.NeutralMode = NeutralModeValue.Brake;
+    cfg.CurrentLimits.StatorCurrentLimit = 0;
```
These current limits are set to 0 and enabled. This will make the motor not run no matter what.

EphraimSonnenberg replied:
🔥🔥🔥🔥

bodenaarhus replied:
100% agree

HarryOTO reviewed (changes requested):
You made CTRE last year look good. Muppet. Jk just a few touch ups ahahaha

KeeganRubingh committed:
Testing PR commit stuff lmao

KeeganRubingh committed:
Reverted github testing stuff

EphraimSonnenberg committed:
Made some code revisions for PR
Fixed MotorID's in real
Made ElevatorIOTalonFX's MotorID's better understanding
Made FingeysIOTalonFX's MotorID's better understanding

KeeganRubingh committed:
Changed sim/real order in robotcontainer and added comments to fingeys and toesies

EphraimSonnenberg reviewed (commented):
Fixed Stuff

EphraimSonnenberg committed:
Removed Outdated Arm Command
Added some unit comments
Removed redundant conversions
Removed unnecessary constant files
Removed unused constants
Added proper current limits

KeeganRubingh commented:
@WasteOfTheOcean review this lmao

HarryOTO reviewed (approved):
Nice fixes
